src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -257,11 +289,37 @@ def create_key(self, arg0, arg1=None, arg2=None, names='i,j,k'):
             a = K(v[0])
             b = K(v[1])
 
-        # QuaternionAlgebra(K, a, b)
         else:
+            # QuaternionAlgebra(K, primes, inv_archimedean)
             K = arg0
-            a = K(arg1)
-            b = K(arg2)
+            if category(K) is not NumberFields():
+                raise ValueError("quaternion algbera must be defined over a number field")
+            if isinstance(arg1, list) and isinstance(arg2, list):
+                if not set(arg2).issubset(set([0,QQ(1/2)])):
+                    raise ValueError("list of local invariants specifying ramification should contain only 0 and 1/2")
+                arg1 = list(set(arg1))
+                if not all([p.is_prime() for p in arg1]):
+                    raise ValueError("quaternion algebra constructor requires a list of primes specifying the ramification")
+                if is_RationalField(K):
+                    if len(arg2) > 1 or (len(arg2) == 1 and is_odd(len(arg1) + 2*arg2[0])):
+                        raise ValueError("quaternion algebra over the rationals must have an even number of ramified places")
+                    D = ZZ.ideal_monoid().prod(arg1).gen()
+                    a, b = hilbert_conductor_inverse(D)
+                    a = QQ(a)
+                    b = QQ(b)
+                else:
+                    if len(arg2) != len(K.real_places()):
+                        raise ValueError("must specify ramification at the real places of %s" % K)
+                    if is_odd(len(arg1) + 2 * sum(arg2)):
+                        raise ValueError("quaternion algebra over the rationals must have an even number of ramified places")
+                    fin_places_pari = [I.pari_prime() for I in arg1]
+                    A = pari(K).alginit([2, [fin_places_pari, [QQ(1/2)] * len(fin_places_pari)], arg2], maxord=0)
+                    a = K(A.algsplittingfield().disc()[1])
+                    b = K(A.algb())
```

Comment:
Unfortunately there is a major issue here that @AurelPage correctly hinted at: The order in which the real embeddings are computed in PARI is indeed not the same as the order of embeddings given by `.real_embeddings()`.

Here is an example:
_sage: K.\<a\> = NumberField(-3*x^5 - 11*x^4 - 4*x^3 + 1)
sage: A = QuaternionAlgebra(K, [], [1/2, 0, 1/2])
Quaternion Algebra (-3*a^4 - 14*a^3 - 15*a^2 - 4*a, -8) with base ring Number Field in a with defining polynomial -3*x^5 - 11*x^4 - 4*x^3 + 1_

Using the `ramified_places` code from #37173, one can check that this algebra ramifies at the two embeddings sending $a$ to $-0.738229069310693$ and to $0.467355922387407$. In `K.real_embeddings()` these are the second and the third embeddings, respectively, but the desired output should be ramified at the third embedding as well as the first embedding sending $a$ to $-3.25393276038648$.

This could be fixed by first precomputing the order in which PARI lists the embeddings, and then permuting the list of local Archimedean invariants accordingly. I am not sure if this is feasible, and maybe @AurelPage has a better idea on how to work around this discrepancy between Sage and PARI.

Reply:
@S17A05 It is not difficult to find the bijection. Consider a generator of the field $K$, and take an $\varepsilon$ that separates its real embeddings. Consider the sequences of real embeddings computed by sage and pari respectively, at precision smaller than $\varepsilon$. Sort them both using indirect sort (with the permutations) and compose the permutations in the right way.

However, this procedure should probably not be hard-coded in the middle of the quaternion algebras code, but should be put in a place that ensures it can be reused. Maybe it is already somewhere? I don't know where that should be.

Reply:
> @S17A05 It is not difficult to find the bijection. Consider a generator of the field K, and take an ε that separates its real embeddings. Consider the sequences of real embeddings computed by sage and pari respectively, at precision smaller than ε. Sort them both using indirect sort (with the permutations) and compose the permutations in the right way.
> 
> However, this procedure should probably not be hard-coded in the middle of the quaternion algebras code, but should be put in a place that ensures it can be reused. Maybe it is already somewhere? I don't know where that should be.

I don't think this is implemented yet. I'm not sure if this is the most elegant solution, but I would suggest adding an optional boolean parameter (e.g. called `pari_perm`, set to `False` by default) to the `.real_embeddings()`-method of a number field such that, if the parameter is set to True, the method returns both the real embeddings and a permutation of the indices that correctly re-indexes the embeddings to be in the order they are computed in in PARI.